include/zephyr/arch/arch_inlines.h

```diff
@@ -24,6 +24,16 @@
 #include <zephyr/arch/xtensa/arch_inlines.h>
 #elif defined(CONFIG_RISCV)
 #include <zephyr/arch/riscv/arch_inlines.h>
+#elif defined(CONFIG_NIOS2)
+#include <zephyr/arch/nios2/arch_inlines.h>
+#elif defined(CONFIG_MIPS)
+#include <zephyr/arch/mips/arch_inlines.h>
+#elif defined(CONFIG_ARCH_POSIX)
+#include <zephyr/arch/posix/arch_inlines.h>
+#elif defined(CONFIG_SPARC)
+#include <zephyr/arch/sparc/arch_inlines.h>
+#else
+#error "Unknown Architecture"
```

Comment:
I know i'm a bit late to the party here, but any thoughts on how this is supposed to work with out-of-tree proprietary architectures (such as the one our company is currently using)?

We don't have SMP support so we don't really care about the CONFIG_MP_MAX_NUM_CPUS, but we do care that we now get an "Unknown Architecture" error.

Could we agree to revert the #error change?